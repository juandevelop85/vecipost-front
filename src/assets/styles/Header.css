header {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 30px 10%;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  background-color: white;
  color: #0845ab;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
}

.logoHeader {
  font-size: 2em;
  cursor: pointer;
  order: 1;
  width: 10%;
  padding: 5px 5px 5px 35px;
  margin-right: auto;
}

nav {
  order: 3;
}

.nav__links {
  list-style: none;
}

.nav__links li {
  display: inline-block;
  padding: 0px 20px;
}

.nav__links li a {
  transition: all 0.3s ease 0s;
  text-decoration: none;
  font-size: 16px;
  color: #0845ab;
}

.nav__links li a:hover {
  color: #fea528;
}

/* .dropdown {
  float: left;
  overflow: hidden;
} */

.dropdown .dropbtn {
  transition: all 0.3s ease 0s;
  text-decoration: none;
  font-size: 16px;
  color: #0845ab;
}

.nav__links a:hover,
.dropdown:hover .dropbtn {
  color: #fea528;
}

.dropdown-content {
  display: none;
  right: 10px;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.dropdown-content a {
  float: none;
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content a:hover {
  background-color: #ddd;
}

.dropdown:hover .dropdown-content {
  display: block;
}

.nav__links .icon {
  display: none;
}

@media screen and (max-width: 600px) {
  .nav__links li {
    display: none;
  }
  .nav__links li.icon {
    float: right;
    display: block;
  }
  .logoHeader {
    width: 20%;
  }
  .nav__links li.icon {
    float: right;
    display: block;
  }
}

@media screen and (max-width: 600px) {
  .nav__links.responsive {
    position: relative;
    display: block;
  }
  .nav__links.responsive .icon {
    position: absolute;
    right: 0;
    top: 0;
  }
  .nav__links.responsive li {
    float: none;
    display: block;
    text-align: left;
  }
  .dropdown.responsive {
    position: relative;
  }
  .dropdown.responsive .icon {
    position: absolute;
    right: 0;
    top: 0;
  }
  .dropdown.responsive li {
    float: none;
    display: block;
    text-align: left;
  }
}
