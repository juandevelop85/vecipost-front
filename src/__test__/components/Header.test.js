import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/header';
import ProviderMock from '../../__mocks__/providerMock';

describe('<Header />', () => {
  test('Render del componente Header', () => {
    const header = shallow(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.length).toEqual(1);
  });
  
  test('Render del Header', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.find('.dropbtn').text()).toEqual('Opciones');
  });
});

describe('Header snapshot', () => {
  test('Comprobar la UI del componente Footer', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
