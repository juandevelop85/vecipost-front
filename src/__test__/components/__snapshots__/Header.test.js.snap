// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot Comprobar la UI del componente Footer 1`] = `
<header>
  <i
    className="fa fa-commenting logoHeader"
    onClick={[Function]}
  />
  <nav>
    <ul
      className="nav__links"
      id="myTopnav"
    >
      <li>
        <a
          href="/create/post"
          onClick={[Function]}
        >
          Publicar post
        </a>
      </li>
      <li>
        <div
          className="dropdown"
        >
          <div
            className="dropbtn"
          >
            Opciones
            <i
              className="fa fa-caret-down"
            />
          </div>
          <div
            className="dropdown-content"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Mi perfil
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Mensajes
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Grupos
            </a>
          </div>
        </div>
      </li>
      <li
        className="icon"
        onClick={[Function]}
      >
        <i
          className="fa fa-bars"
        />
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Header snapshot Comprobar la UI del componente Footer 1`] = `
<header>
  <i
    className="fa fa-commenting logoHeader"
    onClick={[Function]}
  />
  <nav>
    <ul
      className="nav__links"
      id="myTopnav"
    >
      <li>
        <a
          href="/create/post"
          onClick={[Function]}
        >
          Publicar post
        </a>
      </li>
      <li>
        <div
          className="dropdown"
        >
          <div
            className="dropbtn"
          >
            Opciones
            <i
              className="fa fa-caret-down"
            />
          </div>
          <div
            className="dropdown-content"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Mi perfil
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Mensajes
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Grupos
            </a>
          </div>
        </div>
      </li>
      <li
        className="icon"
        onClick={[Function]}
      >
        <i
          className="fa fa-bars"
        />
      </li>
    </ul>
  </nav>
</header>
`;
